sdk/nodejs/appservice/zMixins.ts

```diff
@@ -361,25 +361,57 @@ async function produceDeploymentArchiveAsync(args: MultiCallbackFunctionAppArgs)
     return new pulumi.asset.AssetArchive(map);
 }
 
-function combineAppSettings(args: MultiCallbackFunctionAppArgs): pulumi.Output<{[key: string]: string}> {
+function combineFunctionAppSettings(args: MultiCallbackFunctionAppArgs): pulumi.Output<{[key: string]: string}> {
     const applicationSetting = args.appSettings || {};
     const perFunctionSettings = args.functions !== undefined ? args.functions.map(c => c.appSettings || {}) : [];
-    return pulumi.all([applicationSetting, ...perFunctionSettings]).apply(items => items.reduce((a, b) => ({ ...a, ...b }), {}));
+    return combineAppSettings([applicationSetting, ...perFunctionSettings]);
 }
 
 function redirectConsoleOutput<C extends Context<R>, E, R extends Result>(callback: Callback<C, E, R>) {
-    return (context: C, event: E) => {
+    return (context: C, event: E, ...inputs: any[]) => {
         // Redirect console logging to context logging.
         console.log = context.log;
         console.error = context.log.error;
         console.warn = context.log.warn;
         // tslint:disable-next-line:no-console
         console.info = context.log.info;
 
-        return callback(context, event);
+        return callback(context, event, ...inputs);
     };
 }
 
+/**
+ * Azure Function Binding with the required corresponding application settings (e.g., a connection string setting).
+ */
+export interface BindingSettings {
+    /**
+     * An input or output binding definition.
+     */
+    readonly binding: pulumi.Input<BindingDefinition>;
+
+    /**
+     * A dictionary of application settings to be applied to the Function App.
+     */
+    readonly settings: pulumi.Input<{ [key: string]: any; }>;
+}
+
+// We might want to merge this into CallbackArgs hierachy when all function types support bindings
+export interface InputOutputsArgs {
+    /**
+     * Input and/or Output bindings.
+     */
+    inputOutputs?: appservice.BindingSettings[];
+}
+
+/**
+ * Type alias for a response coming from an Azure Function callback, which applies to most Function types (HTTP being a notable exception).
+ */
+export type FunctionDefaultResponse =
+    // Nothing is returned if a function has no output bindings
+    void |
+    // A dictionary is returned if a function has output bindings
+    Record<string, any>;

```

Comment:
can we move this into the main description?  Also, can you doc what the keys/values are for this dictionary?

Reply:
Done